plugins/modules/system_high_availability_settings.py

```diff
@@ -0,0 +1,489 @@
+
+    # run php function
+    result = opnsense_utils.run_command(
+        php_requirements=php_requirements,
+        command=php_command,
+    )
+
+    # check for stderr
+    if result.get("stderr"):
+        raise OPNSenseGetInterfacesError("error encountered while getting interfaces")
+
+    # parse list
+    interfaces = dict(
+        (item.strip().split(":"))
+        for item in result.get("stdout").split(",")
+        if item.strip() and item.strip() != "None"
+    )
+
+    # check parsed list length
+    if len(interfaces) < 1:
+        raise OPNSenseGetInterfacesError(
+            "error encountered while getting interfaces, no interfaces available"
+        )
+
+    return interfaces
+
+
+def synchronize_interface(config: OPNsenseModuleConfig, sync_interface: str) -> None:
+    """
+    Handler function for the synchronize_interface setting
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        sync_interface (bool): If synchronize_states is enabled,
+                               it will utilize this interface for communication.
+    """
+    interfaces = {"lo0": "Loopback"}
+    interfaces.update(get_configured_interface_with_descr())
+    for ident, desc in interfaces.items():
+        if sync_interface.lower() in (ident.lower(), desc.lower()):
+            config.set(ident, "synchronize_interface")
+            return
+    raise ValueError(
+        f"'{sync_interface}' is not a valid interface. "
+        + "If the interface exists, ensure it is enabled and also not virtual."
+    )
+
+
+def synchronize_peer_ip(config: OPNsenseModuleConfig, peer_ip: str) -> None:
+    """
+    Handler function for the synchronize_peer_ip setting
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        peer_ip: PFsync will sync to this IP address.
+    """
+    if peer_ip:
+        if not validate_ipv4(peer_ip):
+            raise ValueError(
+                "Setting synchronize_peer_ip has to be a valid IPv4 address"
+            )
+        config.set(value=peer_ip, setting="synchronize_peer_ip")
+    elif not peer_ip and config.get("synchronize_peer_ip") is not None:
+        config.get("hasync").remove(config.get("synchronize_peer_ip"))
+
+
+def remote_system_synchronization(
+    config: OPNsenseModuleConfig,
+    remote_backup_url: Optional[str],
+    username: Optional[str],
+    password: Optional[str],
+) -> None:
+    """
+    Handler function for the settings synchronize_config_to_ip,
+    remote_system_username and remote_system_password.
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        remote_backup_url (Optional[str]): Synchronize your settings to this URL
+        username (Optional[str]): Username for logging in to the remote firewall
+        password (Optional[str]): Password for logging in to the remote firewall
+    """
+    if any((remote_backup_url, username, password)):
```

Comment:
Shouldn't all parameters that are set be checked if they are not null?
Why double brackets?

Reply:
The double brackets are present b/c the function `any()` takes only one parameter, which has to be an iterable. Putting all parameters as is into the function wouldn't work.